### PR TITLE
update(query): Apt Get Install Pin Version Not Defined

### LINKED ISSUE #5117: Query "Apt Get Install Lists Were Not Deleted" for Docker does not detect deletion when seperated by `;` instead of `&&`
### Expected Behavior

When the deletion of the apt get install lists happens this is detected regardless of the way the bash commands are chained.

### Actual Behavior

In cases the commands within a `RUN` instruction are separated by `;` kics does not find the removal of the apt get install lists. 

Example:

```dockerfile
RUN set -eux; \
	apt-get update; \
	apt-get install -y --no-install-recommends package=0.0.0; \
	rm -rf /var/lib/apt/lists/*
```

This is a result of the way the query is currently implemented. The query splits the commands by separating them whenever a `&& ` is found. see: https://github.com/Checkmarx/kics/blob/master/assets/queries/dockerfile/apt_get_install_lists_were_not_deleted/query.rego#L23

### Steps to Reproduce the Problem

1. Take the dockerfile for the official tomcat docker image (https://github.com/docker-library/tomcat/blob/29c3afc3135cf9464fa263762b63e0203a7fe178/10.1/jdk17/openjdk-slim-bullseye/Dockerfile)
2. Run kics 
3. See that "Apt Get Install Lists Were Not Deleted" is reported
4. Confirm that the apt get install list is removed in line 125 of the Dockerfile (https://github.com/docker-library/tomcat/blob/29c3afc3135cf9464fa263762b63e0203a7fe178/10.1/jdk17/openjdk-slim-bullseye/Dockerfile#L125)

### Specifications
-   Version: v1.5.5
-   Platform: Docker
-   Subsystem:  Dockerfile
-   Query: `df746b39-6564-4fed-bf85-e9c44382303c`

### PR DESCRIPTION
**Pull request by cxMiguelSilva**
Closes #5117

**Proposed Changes**
- Update query Apt Get Install Pin Version Not Defined
- Added support for docker ; command separation in the query mentioned above
- Update dockerfile.rego functions to add support for docker ; command separation

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/dockerfile/apt_get_install_lists_were_not_deleted/query.rego [hunk 1]
```diff
@@ -32,5 +32,21 @@ hasClean(resourceValue, aptGet) {
...
+	startswith(listCommands[clean], "apt-get clean")
+
+	install < clean
+} else {
+	res:= replace(resourceValue, "\t", "")
+	listCommands := split(res, "; ")
+
+	startswith(listCommands[install], aptGet)
+	startswith(listCommands[remove], "rm -rf")
+
```
Comment (rafaela-soares) [inline assets/queries/dockerfile/apt_get_install_lists_were_not_deleted/query.rego:50]:
```suggestion
options := {"&& ", "; "}

hasClean(resourceValue, aptGet) {
	res:= replace(resourceValue, "\t", "")
	listCommands := split(res, options[_])

	startswith(listCommands[install], aptGet)
	startswith(listCommands[clean], "apt-get clean")

	install < clean
} else {
	res:= replace(resourceValue, "\t", "")
	listCommands := split(res, options[_])

	startswith(listCommands[install], aptGet)
	startswith(listCommands[remove], "rm -rf")

	install < remove
}
```

Comment (rafaela-soares) [inline assets/queries/dockerfile/apt_get_install_lists_were_not_deleted/query.rego:50]:
```suggestion
options := {"&& ", "; "}

hasClean(resourceValue, aptGet) {
	res:= replace(resourceValue, "\t", "")
	listCommands := split(res, options[_])

	startswith(listCommands[install], aptGet)
	startswith(listCommands[clean], "apt-get clean")

	install < clean
} else {
	res:= replace(resourceValue, "\t", "")
	listCommands := split(res, options[_])

	startswith(listCommands[install], aptGet)
	startswith(listCommands[remove], "rm -rf")

	install < remove
}
```

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

File: assets/queries/dockerfile/apt_get_install_pin_version_not_defined/test/positive2.dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FROM busybox4
```
Comment (rafaela-soares) [inline assets/queries/dockerfile/apt_get_install_pin_version_not_defined/test/positive2.dockerfile:1]: Can you add a negative sample with `;` for `apt_get_install_pin_version_not_defined`, please?

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

**Review by joaorufi — APPROVED** [review]

**Review by joaorufi — DISMISSED** [review]

**Review by joaorufi — APPROVED** [review]

**Review by rafaela-soares — APPROVED** [review]
LGTM 🚀

**Pull request merged**